lib/modules/apostrophe-login/index.js

```diff
@@ -311,6 +313,37 @@ module.exports = {
       });
     };
 
+    self.disableIfInactive = function(err, user, callback) {
+      if (self.apos.users.options.disableInactiveAccounts) {
+        var usersOptions = self.apos.users.options.disableInactiveAccounts;
+
+        if (user.lastLogin) {
+          // by default, "admin" users are not disabled
+          var isUserWhitelisted = user._groups.some(function(group) {
+            return usersOptions.whitelistGroups.includes(group.title);
+          });
+
+          if (!isUserWhitelisted) {
+            var lastLogin = moment(user.lastLogin);
+            var now = moment();
+            var diff = lastLogin.diff(now, 'days');
+
+            if (diff >= usersOptions.inactivityDuration) {
+              var req = self.apos.tasks.getReq();
+              user = Object.assign({}, user, { disabled: true, trash: false, lastLogin: '' });
```

Comment:
Use `apos.docs.db.update`, i.e. use the mongodb docs collection, so you can use `$set` and avoid a possible race condition as well as your Object.assign step.

Reply:
Done